hypothesis-python/src/hypothesis/extra/numpy.py

```diff
@@ -248,7 +258,7 @@ def set_element(self, val, result, idx, *, fill=False):
             # This branch only exists to help debug weird behaviour in Numpy,
             # such as the string problems we had a while back.
             raise HypothesisException(
-                f"Internal error when checking element={val!r} of {val.dtype!r} "
+                f"Internal error when checking element={val!r} of {getattr(val, 'dtype', type(val))!r} "
```

Comment:
It doesn't help that this code isn't typed (not your fault!), but do we expect it be be possible that `val` is not a numpy object and therefore doesn't have `dtype`? If so, we should also change the `possible mismatch of time units in dtypes` code above. It seems this might be a latent bug.

Reply:
Yes this it is now possible that val is not a numpy object so we need to fallback to use `type()`. However, I think we still need the above anyway just in case the wrong `elements` are used in a typed `array`.